tests/benchmark/utils/s2n_negotiate_server_benchmark.cc

```diff
@@ -0,0 +1,304 @@
+static char ecdsa_certificate_chain[] =
+        "-----BEGIN CERTIFICATE-----"
+        "MIICLDCCAdGgAwIBAgIUPYYEnK24qDzz59IIDcNLc4P2H5swCgYIKoZIzj0EAwIw"
+        "XzELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAldBMRAwDgYDVQQHDAdTZWF0dGxlMQ8w"
+        "DQYDVQQKDAZBbWF6b24xDDAKBgNVBAsMA3MybjESMBAGA1UEAwwJbG9jYWxob3N0"
+        "MCAXDTIwMTIwNDA3NDg1NloYDzIxMjAxMTEwMDc0ODU2WjBfMQswCQYDVQQGEwJV"
+        "UzELMAkGA1UECAwCV0ExEDAOBgNVBAcMB1NlYXR0bGUxDzANBgNVBAoMBkFtYXpv"
+        "bjEMMAoGA1UECwwDczJuMRIwEAYDVQQDDAlsb2NhbGhvc3QwWTATBgcqhkjOPQIB"
+        "BggqhkjOPQMBBwNCAARhONnk1k68YnnabiHzf3AvlWwN93SOvdq6v1Grl3YEiGM1"
+        "W8WFH7O4cxb+otlVlhhbPzaox4EVthLExJZumx8go2kwZzAdBgNVHQ4EFgQU0ip8"
+        "rN6YlbtCUIueCOqfh3/J3KMwHwYDVR0jBBgwFoAU0ip8rN6YlbtCUIueCOqfh3/J"
+        "3KMwDwYDVR0TAQH/BAUwAwEB/zAUBgNVHREEDTALggkxMjcuMC4wLjEwCgYIKoZI"
+        "zj0EAwIDSQAwRgIhAJwlrxN5SDi2dC17ZPgajqZ8BZyOsNFE+gsobhMBUGN0AiEA"
+        "6KFJgyPGBNdQqaczkNyBcutPGqEubuah5Me6faN4qqU="
+        "-----END CERTIFICATE-----";
+
+//ecdsa_p384_pkcs1_key.pem
+static char ecdsa_private_key[] =
+        "-----BEGIN EC PARAMETERS-----"
+        "BggqhkjOPQMBBw=="
+        "-----END EC PARAMETERS-----"
+        "-----BEGIN EC PRIVATE KEY-----"
+        "MHcCAQEEIK4AEDQja7MDATqWWu4T0+iMFdSZH4y4+nuVzDX5ao8KoAoGCCqGSM49"
+        "AwEHoUQDQgAEYTjZ5NZOvGJ52m4h839wL5VsDfd0jr3aur9Rq5d2BIhjNVvFhR+z"
+        "uHMW/qLZVZYYWz82qMeBFbYSxMSWbpsfIA=="
+        "-----END EC PRIVATE KEY-----";
+
+static int server_benchmark(benchmark::State& state, bool warmup) {
+    int fd = fd_bench;
+    struct s2n_config *config = config_once;
+    struct conn_settings settings = conn_settings;
+    struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+    if (!conn) {
+        print_s2n_error("Error getting new s2n connection");
+        S2N_ERROR_PRESERVE_ERRNO();
+    }
+
+    s2n_setup_server_connection(conn, fd, config, settings);
+
+    if (benchmark_negotiate(conn, fd, state, warmup) != S2N_SUCCESS) {
+        if (settings.mutual_auth) {
+            if (!s2n_connection_client_cert_used(conn)) {
+                print_s2n_error("Error: Mutual Auth was required, but not negotiated");
+            }
+        }
+
+        S2N_ERROR_PRESERVE_ERRNO();
+    }
+
+    GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
+
+    s2n_blocked_status blocked;
+    s2n_shutdown(conn, &blocked);
+
+    GUARD_RETURN(s2n_connection_wipe(conn), "Error wiping connection");
+
+    GUARD_RETURN(s2n_connection_free(conn), "Error freeing connection");
+
+    return 0;
+}
+
+static int ServerBenchmark(benchmark::State& state) {
+    int i;
+    for(i = 0; i < WARMUP_ITERS; i++) {
+        server_benchmark(state, true);
+    }
+    for(auto _ : state) {
+        state.PauseTiming();
+        server_benchmark(state, false);
+    }
+    return 0;
+}
+
+
+int Server::start_benchmark_server(int argc, char **argv) {
+    struct addrinfo hints, *ai;
+    int r, sockfd= 0;
```

Comment:
Is `r` used anywhere? Also please try to avoid single character variable names, especially for large functions like these.